sdk/identity/Azure.Identity/src/Credentials/AzurePipelinesServiceConnectionCredential.cs

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Core.Pipeline;
+using Microsoft.Identity.Client;
+
+namespace Azure.Identity
+{
+    /// <summary>
+    /// Credential which authenticates using an Azure Pipelines service connection.
+    /// </summary>
+    public class AzurePipelinesServiceConnectionCredential : TokenCredential
+    {
+        internal readonly string[] AdditionallyAllowedTenantIds;
+        internal string TenantId { get; }
+        internal string ClientId { get; }
+        internal MsalConfidentialClient Client { get; }
+        internal CredentialPipeline Pipeline { get; }
+        internal TenantIdResolverBase TenantIdResolver { get; }
+        private const string OIDC_API_VERSION = "7.1";
+
+        /// <summary>
+        /// Protected constructor for mocking.
+        /// </summary>
+        protected AzurePipelinesServiceConnectionCredential()
+        { }
+
+        /// <summary>
+        /// Creates a new instance of the <see cref="AzurePipelinesServiceConnectionCredential"/>.
+        /// </summary>
+        /// <param name="tenantId">The tenant Id for the service connection.</param>
+        /// <param name="clientId">The client Id for the service connection.</param>
+        /// <param name="serviceConnectionId">The service connection Id.</param>
```

Comment:
Will most customers know how to find the service connection ID? If there's a doc we can link to for this param, that would be helpful.
```suggestion
        /// <param name="tenantId">The tenant ID for the service connection.</param>
        /// <param name="clientId">The client ID for the service connection.</param>
        /// <param name="serviceConnectionId">The service connection ID.</param>
```